### ISSUE TITLE
Nothing Loads on the Filter Page

### ISSUE THREAD

**Issue by busterroni** [teshiron/davt#21]
Hey,

The page is loaded and I can see the 'Pause updates' button along with all of the other Wikipedia stuff, but no filtered edits are loading. I've been waiting for about 3 minutes.

Another bug, I cannot click on 'Talk', 'Edit', or 'View history'. The page I clicked on loads for a half second then forwards back to the content from the main page (the 'Read' page). However, it still says I'm on the page I clicked on. Here's a picture:
![screen shot 2013-11-06 at 10 53 04 pm](https://f.cloud.github.com/assets/3424032/1489005/2693c67a-4760-11e3-8350-f97cb3380b73.png)

As you can see, I'm on the 'View history' page, but I'm given the content from the 'Read' page!

Another error is that after I click on 'Pause updates' clicking on 'Resume updates' changes nothing. It still says 'Resume updates' after I click on it!

I'm on a Mac OS X Lion running Version 10.7.5 and Chrome Version 30.0.1599.101.

Thanks!
busterroni (on Wikipedia I'm Newyorkadam)

P.S. Also, where do you get your bad word list from? I was checking your code and it loads from somewhere, I'm just not sure where. Can I see a link please?

P.P.S. I also get a HUGE list of errors in my console, and the list keeps growing and growing with about seven more errors per second. Here's some of the errors (in order from when the page loaded: the first error is when the page first loaded):

Entering filterChanges index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:39
/w/api.php?action=query&list=recentchanges&format=json&rcdir=older&rcprop=ids&rclimit=100&rctoponly=&rcshow=!minor|!bot&rctype=edit|new&rcnamespace=0|2&rcstart=2013-11-07T03:47:03.788Z index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:81
Entering RC ajax success function index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:87
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:143
Revision is 580548585 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:163
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:167
NP Queue is empty index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:170
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 98 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548584 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 97 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548583 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 96 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548579 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 95 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548578 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 94 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548576 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 93 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548572 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 92 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548570 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 91 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548569 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 90 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548567 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 89 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548566 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 88 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548564 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 87 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548563 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 86 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548562 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 85 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548561 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 84 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548559 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 83 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548557 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
Match found index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:287
Diff queue length is: 82 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:307
Uncaught TypeError: Cannot call method 'slice' of undefined index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:384
Revision is 580548550 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 81 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548542 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 80 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548540 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 79 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548539 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 78 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548538 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 77 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548537 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236
Testing for a match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:259
No match index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:276
Diff queue length is: 76 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:282
Revision is 580548534 index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:211
Still latest, pulling full content index.php?title=User:Darkwind/DAVT.js&action=raw&ctype=text/javascript:236

**Comment by teshiron** [comment]
Let me try to respond in order:

It looks like it just might not have found any matches to show.  Each time it says "no match" in the console log, it means it tested a diff and didn't find any bad words to match against.  Depending on the time of day, it might just take a long time to show anything.  The fact that it's showing all of those things means the script is functioning, and just not finding anything.

As for the talk/history, that's because the script actually loads on every Wikipedia page, and starts processing and displaying diffs when you're on any page where the URL contains "DAVT/Filter".  You'll notice the same behavior in Lupin's tool when you go to the Filter Recent Changes page, then click View History.

There's nothing to see on the history page anyway (the page has one edit which is an HTML comment to generate a blank page).  If you think it would be helpful, I can have the script put a link to the correct talk page at the top above "pause updates".  Alternatively, I might be able to change the filter page to a variation of Special:BlankPage, like the RTRC gadget does, so it's clearer to users that there is no content to see on talk/history/etc.

The "errors" you pasted are actually just the script logging its actions (mostly for debugging purposes).  In the JavaScript console in Chrome, you can hold control (or possibly command? dunno Macs very well) and click "Logs" at the bottom, and it will suppress the log messages and only show actual errors.  It's the same key-click-combo you use to select multiple items from a list, whatever that is on a Mac.

Regarding the pause/resume function, it definitely works in the version I'm ready to release tonight or tomorrow, or at least it does in Firefox on a PC.  After I update the tool to 0.1.0 tonight or tomorrow, please try the pause/resume functionality again, and let me know if it still doesn't work -- and if it doesn't, see if there are any errors showing in the JavaScript console when you click pause or resume.

The badwords list is basically the same as the one used for Lupin's tool, but I removed all of the image-specific strings (I hadn't seen Lupin's tool match on an image file name in years).  It's at [[User:Darkwind/DAVT/badwords]].  The list for Lupin's tool is at [[User:Lupin/badwords]].

**Comment by busterroni** [comment]
Hey there, I see on [the Wikipedia article](https://en.wikipedia.org/wiki/User:Darkwind/DAVT) that the last stable release was published today (11/11/13). However, the Anti-Vandal tool still says "This page was last modified on 29 October 2013 at 22:56." The pause button still functions as it did before (I click "Pause updates" and it shows "Resume updates", however I can't click back to "Pause updates").

On the Chrome console:

Here's the 'Logs' page:
Still latest, pulling full content index.php:183
Revision is 581216556 index.php:256
Still latest, pulling full content index.php:279
Testing for a match index.php:202
Downloading editor properties index.php:330
Editor whitelisted index.php:358
Diff queue length is: 94 index.php:364
Revision is 581216555 index.php:256
Still latest, pulling full content index.php:279
Downloading editor properties index.php:330
Editor whitelisted index.php:358
Diff queue length is: 93 index.php:364
Revision is 581216554 index.php:256
Still latest, pulling full content index.php:279
Downloading editor properties index.php:330
Testing for a match index.php:369

Here's the 'Errors' page:
Uncaught TypeError: Cannot call method 'test' of null index.php:205
Uncaught TypeError: Cannot call method 'test' of null index.php:379

I was about to write that it wasn't working but the AHAH! It started working, sick :D

Thanks :)
-busterroni

**Comment by teshiron** [comment]
The page [[User:Darkwind/DAVT.js]] is what gets updated when I make changes to the tool/code. The page [[User:Darkwind/DAVT/Filter]] does not get edited, like, ever, so its last update time will never change.  

As for the pause/resume thing, the button is functional - if you watch the logs, when you click pause, it stops, and when you click resume, it resumes -- but the text of the button does not appear to change; that's a bug. I'll get a fix out shortly.

The uncaught typeErrors are confusing -- there's no reason codewise why that should be happening.  I encountered the same errors, and when I refreshed the page, they didn't happen the next time.  It may have something to do with how long it takes Wikipedia to return the page containing the bad word list.  If it times out, it may cause those errors.
